FileShareClient/AuthTokenProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Threading.Tasks;
+
+namespace UKHO.FileShareClient
+{
+    public interface IAuthTokenProvider
+    {
+        Task<string> GetToken();
+    }
+
+    internal class DefaultAuthTokenProvider: IAuthTokenProvider
```

Comment:
Should we rename the file to match the primary class in here: i.e. `DefaultAuthTokenProvider.cs`?